src/main/scala/analysis/Analysis.scala

```diff
@@ -500,206 +495,149 @@ trait MemoryRegionAnalysisMisc:
 
   private val mallocVariable = Register("R0", BitVecType(64))
 
-  private val loopEscapeSet: mutable.Set[CfgNode] = mutable.Set.empty
-
-  def loopEscape(n: CfgNode): Boolean = {
-    if (loopEscapeSet.contains(n)) {
-      return true
-    }
-    loopEscapeSet.add(n)
-    false
-  }
-
-  /** Find decl of variables from node predecessors */
-  def findDecl(variable: Variable, n: CfgNode): mutable.ListBuffer[CfgNode] = {
-    val decls: mutable.ListBuffer[CfgNode] = mutable.ListBuffer.empty
-    // if we have a temporary variable then ignore it
-    variable match {
-      case r: Register =>
-        for (pred <- n.pred) {
-          if (loopEscape(pred)) {
-            return mutable.ListBuffer.empty
+  /** Default implementation of eval.
+    */
+  def eval(exp: Expr, env: lattice.sublattice.Element, n: CfgNode): lattice.sublattice.Element = {
+    Logger.debug(s"evaluating $exp")
+    Logger.debug(s"env: $env")
+    Logger.debug(s"n: $n")
+    exp match {
+      case binOp: BinaryExpr =>
+        if (binOp.arg1 == stackPointer) {
+          val rhs: Expr = evaluateExpression(binOp.arg2, n, constantProp)
```

Comment:
Pointless line?

Reply:
Resolved. [Upcoming merge from yousif-memory-region-analysis]
rhs should be passed instead of binOp.arg2